addons/stock/models/stock_lot.py

```diff
@@ -213,6 +213,32 @@ def _search_product_qty(self, operator, value):
             return ['|', ('id', 'in', ids), ('id', 'not in', lot_ids_w_qty)]
         return [('id', 'in', ids)]
 
+    def _search_last_delivery_partner_id(self, operator, value):
+        lot_ids = []
+        picking_ids = self.env['stock.picking'].search([
+            ('state', '=', 'done'),
+            ('product_id.tracking', '=', 'serial'),
+            ('picking_type_id', '=', self.env.ref('stock.picking_type_out').id)
+        ]).sorted('date_done', reverse=True)
+        if picking_ids:
+            lot_ids = picking_ids.move_line_ids.lot_id if operator in ['!=', '='] else picking_ids[0].move_line_ids.lot_id
+        res_partner = self.env['res.partner']
+
+        if isinstance(value, bool) and operator == '!=':
+            return [('id', 'in', lot_ids.ids)]
+        elif isinstance(value, bool) and operator == '=':
+            return [('id', 'not in', lot_ids.ids)]
+        elif isinstance(value, (int, list)):
+            partner_ids = res_partner.search([('id', operator, value)])
+        else:
+            partner_ids = res_partner.search([('name', operator, value)])
+        domain = [('id', 'in', [])]
+        if not partner_ids:
+            return domain
+        if picking_ids:
+            domain = [('id', 'in', lot_ids.ids)] if picking_ids[0].mapped('partner_id') in list(partner_ids) else domain
+        return domain
```

Comment:
This is better than your previous attempt (next time please either make your commit a `fixup` commit or squash it with your old commit. I spent a lot of time looking at your old implementation :cry: ).

It can use a couple of improvements though:
- create the domain in your if/ifel statement so that if the `partner_ids` are part of the search, then you will only need to do 1 search on `stock.picking`
- you should add some checks to make sure the `operator` and `value` are valid in case someone does a direct call of the method
- there is some extra logic in `_find_delivery_ids_by_lot` to capture lots that have been used in an MO (i.e. consumed) that links the lot to a delivery of the produced product from the MO. It would be better if you can also capture this logic, but there's a chance the existing logic isn't working correct (it doesn't look like it is based on my functional testing, so we can maybe skip it for now)

Reply:
@ticodoo, Thank you for your detail review. I believe this particular improvement - searching lots based on last delivery partner_id isn't required at this point, as the current stock lot view no longer incorporates delivery lots. Consequently, searching for the last delivery partner ID seems redundant.

Reply:
@rare-odoo Thanks for letting me know! It's a mistake that the delivered lots aren't shown in the stock lot view anymore :see_no_evil: We will do a fix so that they are shown again, which means the `last_delivery_partner_id` search is still needed (lucky for you :partying_face: )

Reply:
Indeed 😅